This update adds support for the OEM Series controller

guirodrigueslima opened the pull request:
This pull request adds support for the OEM Series controller. Its implementation provides basic motion control. In addition, it allows selecting a limit switch (CW or CCW) to reset the absolute position to zero and also checks whether a limit switch was detected. 

The development structure is based on the ACR Series controller.

guirodrigueslima committed:
Start development for the OEM Series controller.

henriquesimoes commented on parkerApp/src/OEMMotorDriver.cpp:
```diff
@@ -0,0 +1,411 @@
+  return status;
+}
+
+asynStatus OEMAxis::moveVelocity(double minVelocity, double maxVelocity, double acceleration)
+{
+  asynStatus status = asynSuccess;
+  static const char *functionName = "moveVelocityAxis";
+
+  return status;
+}
```
Is this correct?

henriquesimoes commented on parkerApp/src/OEMMotorDriver.cpp:
```diff
@@ -0,0 +1,411 @@
+  return status;
+}
+
+asynStatus OEMAxis::setPosition(double position)
+{
+  asynStatus status = asynSuccess;
+  static const char *functionName = "setPositionAxis";
+  return status;
+}
```
Here as well.

henriquesimoes commented on parkerApp/src/Makefile:
```diff
@@ -15,17 +15,23 @@ include $(TOP)/configure/CONFIG
 DBD += ParkerSupport.dbd
 
 # Build a library wtih only ACR support for backwards compatibility
 LIBRARY_IOC += ACRMotor
+LIBRARY_IOC += OEMMotor
 
 ACRMotor_SRCS += ACRMotorDriver.cpp
-
 ACRMotor_LIBS += motor
 ACRMotor_LIBS += asyn
 ACRMotor_LIBS += $(EPICS_BASE_IOC_LIBS)
 
+OEMMotor_SRCS += OEMMotorDriver.cpp
+OEMMotor_LIBS += motor
+OEMMotor_LIBS += asyn
+OEMMotor_LIBS += $(EPICS_BASE_IOC_LIBS)
+
```
You don't need to provide a library for OEM here, as there is no need for backward compatibility in this case. No one depends on it. They should depend on Parker directly.

henriquesimoes commented on parkerApp/src/OEMMotorDriver.cpp:
```diff
@@ -0,0 +1,411 @@
+FILENAME... OEMMotorDriver.cpp
+USAGE...    Motor driver support for the Parker OEM series of controllers, including the Aries.
+
+Mark Rivers
+March 4, 2011
```
This copyright is not correct.

henriquesimoes commented on parkerApp/src/OEMMotorDriver.h:
```diff
@@ -0,0 +1,66 @@
+FILENAME...   OEMMotorDriver.h
+USAGE...      Motor driver support for the Parker OEM series of controllers, including the Aries.
+
+Mark Rivers
+March 28, 2011
```
Here as well.

henriquesimoes reviewed (commented):
Just a quick review done while we were discussing in person.

guirodrigueslima commented:
Thanks @henriquesimoes, I applied your suggested change.

guirodrigueslima commented:
ping

guirodrigueslima committed:
Support motor resolution and configure controller

- The MRES resolution can be defined according to the system applied (e.g., rev, mm, etc.).
- Motor revolution (step/rev) need to be writting in OEMSupport.template
- The configure controller function was created to allow the controller to be reconfigured in case it is turned off and on again.

guirodrigueslima committed:
Motor status was included
